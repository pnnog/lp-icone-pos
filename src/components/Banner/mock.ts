export default {
  img: 'https://source.unsplash.com/user/willianjusten/1040x495',
  reminder: true,
  title: 'Especialização implantodontia',
  date: '1/1/1',
  teacher: 'Dr. Sandro Bittencourt | Dr. Márcio Marchionni'
};
