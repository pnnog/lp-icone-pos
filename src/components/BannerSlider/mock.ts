export default [
  {
    reminder: true,
    title: 'Especialização implantodontia',
    img: 'https://source.unsplash.com/user/willianjusten/1600x800',
    date: '25/03/2023',
    teacher: 'Dr. Sandro Bittencourt | Dr. Márcio Marchionni'
  },

  {
    title: 'Aperfeiçoamento em implantodontia',
    img: 'https://source.unsplash.com/user/willianjusten/1440x496',
    date: '29/03/2023',
    teacher: 'Dr. Sandro Bittencourt | Dr. Márcio Marchionni'
  }
];
