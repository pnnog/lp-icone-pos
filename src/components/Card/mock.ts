export const mockPacientCards = [
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x400',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tempor pradadsasoidsent exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  },

  {
    img: 'https://source.unsplash.com/user/willianjusten/400x401',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tdadasdempor proideaasdnt exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  },
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x402',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tdadasdempor pdddroident exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  },
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x403',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tsadasdadadddadasdempor proident exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  },
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x404',
    title: 'Irure ad reprehenderit dolore voluptate Lorem adasdasdasdtdadasdempor proident exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  }
];

export const mockCourseCards = [
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x400',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tempor proident exercitation qsdasdasdui.'
  },

  {
    img: 'https://source.unsplash.com/user/willianjusten/400x401',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tempor proident exercitadasdasdation qui.'
  }
];
