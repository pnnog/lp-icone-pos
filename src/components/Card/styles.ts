import styled, { css } from 'styled-components';
import media from 'styled-media-query';

export const Wrapper = styled.div`
  width: 100%;
  height: 100%;
`;

export const Card = styled.div`
  ${({ theme }) => css`
    border-radius: ${theme.border.radius};

    box-shadow: 1.8px 3.6px 1.8px rgba(0, 0, 0, 0.25);
    position: relative;
    border: none;
    width: 100%;
    height: 100%;
    cursor: pointer;
  `}
`;

export const ImageBox = styled.figure`
  width: 100%;
  height: 25rem;
  border-radius: inherit;

  ${media.greaterThan('medium')`
    height: 30rem;
  `}

  background: #f6f7f8;
  background-image: linear-gradient(
    to right,
    #f6f7f8 0%,
    #edeef1 20%,
    #f6f7f8 40%,
    #f6f7f8 100%
  );
  background-size: 80rem 14rem;
  animation: placeholderShimmer 1s linear infinite forwards;
  @keyframes placeholderShimmer {
    0% {
      background-position: -40rem 0;
    }
    100% {
      background-position: 40rem 0;
    }
  }
`;
export const Image = styled.img`
  width: 100%;
  height: 100%;
  border-radius: inherit;
`;

export const Content = styled.div`
  position: absolute;
  bottom: 2rem;
  left: 50%;
  transform: translateX(-50%);
  width: 85%;
  word-wrap: break-word;
`;
export const Title = styled.h2`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.small};
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.white};
    text-shadow: 0 0.3rem 0.6rem rgba(0, 0, 0, 0.4);
    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xlarge};
    `}
  `}
`;

export const More = styled.div`
  ${({ theme }) => css`
    position: relative;
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
    border-radius: ${theme.border.radius};
    text-align: center;

    border-radius: ${theme.border.radius};
    backdrop-filter: blur(1.5rem);
    background: rgba(2, 132, 199, 0.7);
    box-shadow: 0 0 1rem 01rem rgba(0, 0, 0, 0.35);
    -webkit-backdrop-filter: blur(1.5rem);

    span {
      color: ${theme.colors.white};
      font-weight: ${theme.font.bold};
      font-size: ${theme.font.sizes.small};
    }

    ${media.greaterThan('medium')`
      padding: ${theme.spacings.xsmall} ${theme.spacings.medium};

      span{
        font-size: ${theme.font.sizes.large};
      }
    
    `}
  `}
`;

const ModalModifiers = {
  isOpen: () => css`
    opacity: 1;
    pointer-events: all;
  `,
  isClose: () => css`
    opacity: 0;
    pointer-events: none;
  `
};

type ModalProps = {
  isOpen?: boolean;
};

export const Modal = styled.div<ModalProps>`
  ${({ theme, isOpen }) => css`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: ${theme.layers.modal};
    overflow-y: auto;
    overflow-x: hidden;

    display: flex;
    padding-left: ${theme.spacings.xsmall};
    padding-right: ${theme.spacings.small};

    backdrop-filter: blur(1.5rem);
    -webkit-backdrop-filter: blur(1.5rem);
    background: rgba(0, 0, 0, 0.7);

    transition: opacity 0.3s ease-out;

    ${!!isOpen && ModalModifiers.isOpen()}
    ${!isOpen && ModalModifiers.isClose()};

    ${media.greaterThan('medium')`
      display:flex;
      justify-content: center;
      background: rgba(12, 74, 110, .8);
    `}
  `}
`;

export const Close = styled.div`
  ${({ theme }) => css`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    cursor: pointer;

    color: ${theme.colors.white};
    text-align: right;

    svg {
      margin: ${theme.spacings.xxsmall};

      ${media.greaterThan('large')`
        margin: ${theme.spacings.small};
      `}
    }
  `}
`;

export const ModalContent = styled.div`
  ${({ theme }) => css`
    width: min(100%, 90rem);
    margin-top: ${theme.spacings.small};

    ${media.greaterThan('medium')`
      margin-top: ${theme.spacings.xxlarge};
    `}

    ${Title} {
      text-shadow: none;
      position: relative;
      margin-bottom: calc(${theme.spacings.small}*2);

      ::after {
        content: '';
        position: absolute;
        width: 100%;
        height: 0.6rem;
        bottom: -2rem;
        left: 0;
        background-color: ${theme.colors.secondary};
      }

      ${media.greaterThan('medium')`
        margin-bottom: ${theme.spacings.xxlarge};

        ::after{
          background-color: ${theme.colors.white};
        }
      `}
    }
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.white};
    padding-bottom: ${theme.spacings.small};
    text-align: justify;

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}
  `}
`;
