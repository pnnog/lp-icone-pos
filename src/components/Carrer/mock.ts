export default {
  img: '/img/gustavo-bastos.jpg',
  name: 'Gustavo Bastos',
  role: 'Cordenador',
  description:
    'Cillum voluptate consectetur nostrud proident deserunt culpa laboris incididunt quis sunt cillum. Dolor aliquip sunt pariatur ullamco labore labore velit sit dolore mollit elit proident. Amet consequat quis ipsum anim deserunt eiusmod. Tempor qui anim cillum voluptate nulla id minim sit. Consectetur irure ex cupidatat officia laboris dolor id pariatur fugiat ut voluptate non. Ea reprehenderit tempor officia eiusmod ex laborum amet incididunt ea culpa ullamco ad ipsum ut. Quis elit sint proident reprehenderit consectetur Lorem incididunt occaecat voluptate proident.'
};
