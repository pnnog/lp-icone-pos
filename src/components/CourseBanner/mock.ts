export default {
  img: 'https://source.unsplash.com/user/willianjusten/1000x1000',
  title: 'Especialização em implantodontia',

  beginning: {
    title: 'Início',
    description: 'Março de 2023'
  },

  workload: {
    title: '1000 h/a',
    description: '26 módulos mensais'
  }
};
