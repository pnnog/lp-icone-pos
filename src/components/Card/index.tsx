import { useState } from 'react';
import * as S from './styles';
import { Close } from '@styled-icons/material-outlined';

export type CardProps = {
  img: string;
  title: string;
  modal?: string;
};

const Card = ({ img, title, modal }: CardProps) => {
  const [isOpen, setIsOpen] = useState(false);

  const modalVerifiy = (modal: string | undefined) => {
    !!modal && setIsOpen(true);
    return;
  };

  return (
    <S.Wrapper>
      <S.Card role="button" aria-label="open modal" onClick={() => modalVerifiy(modal)}>
        <S.ImageBox>
          <S.Image src={img} alt={title} />
        </S.ImageBox>
        <S.Content>
          <S.Title>{title}</S.Title>
          <S.More>
            <span> Saiba mais</span>
          </S.More>
        </S.Content>
      </S.Card>

      <S.Modal aria-label="modal" aria-hidden={!isOpen} isOpen={isOpen}>
        <S.Close role="button" aria-label="close modal" onClick={() => setIsOpen(false)}>
          <Close size={30} />
        </S.Close>

        <S.ModalContent>
          <S.Title>{title}</S.Title>
          <S.Description>{modal}</S.Description>
        </S.ModalContent>
      </S.Modal>
    </S.Wrapper>
  );
};

export default Card;
