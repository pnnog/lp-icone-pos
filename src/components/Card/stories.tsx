import { Meta, Story } from '@storybook/react';
import { mockPacientCards } from './mock';
import Card, { CardProps } from '.';

export default {
  title: 'Card',
  component: Card,
  args: { ...mockPacientCards[0] },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta;

export const Basic: Story<CardProps> = (args) => (
  <div style={{ maxWidth: '40rem', margin: 'auto', paddingTop: '10rem' }}>
    <Card {...args} />
  </div>
);
