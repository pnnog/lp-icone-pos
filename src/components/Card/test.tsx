import { fireEvent, screen } from '@testing-library/react';
import { mockPacientCards } from './mock';

import Card from '.';
import { renderWithTheme } from 'utils/tests/helper';

describe('<Card />', () => {
  it('should render Card with title and img', () => {
    renderWithTheme(<Card {...mockPacientCards[0]} />);

    expect(
      screen.getByRole('heading', { name: mockPacientCards[0].title })
    ).toBeInTheDocument();

    expect(
      screen.getByRole('img', { name: mockPacientCards[0].title })
    ).toHaveAttribute('src', mockPacientCards[0].img);
  });

  it('should render modal open', () => {
    renderWithTheme(<Card {...mockPacientCards[0]} />);

    //seleciona o modal
    const modal = screen.getByLabelText('modal');

    //verifica se tá escondido
    expect(modal).toHaveAttribute('aria-hidden', 'true');
    expect(modal).toHaveStyle({ opacity: 0, pointerEvents: 'none' });

    //seleciona o botão de abrir o modal
    const openModalButton = screen.getByRole('button', { name: /open modal/i });

    //abre o modal
    fireEvent.click(openModalButton);

    //verifica se tá aberto
    expect(modal).toHaveAttribute('aria-hidden', 'false');
    expect(modal).toHaveStyle({ opacity: 1, pointerEvents: 'all' });
  });

  it('should render modal close', () => {
    renderWithTheme(<Card {...mockPacientCards[0]} />);

    //seleciona o modal
    const modal = screen.getByLabelText('modal');

    //seleciona o botão de abrir o modal
    const openModalButton = screen.getByRole('button', { name: /open modal/i });

    //abre o modal
    fireEvent.click(openModalButton);

    //seleciona o botão de fechar o modal
    const closeModalButton = screen.getByRole('button', {
      name: /close modal/i
    });

    //fecha o modal
    fireEvent.click(closeModalButton);

    //verifica se tá escondido
    expect(modal.getAttribute('aria-hidden')).toBe('true');
    expect(modal).toHaveStyle({ opacity: 0, pointerEvents: 'none' });
  });

  it('should render modal content', () => {
    renderWithTheme(<Card {...mockPacientCards[0]} />);
    const openModalButton = screen.getByRole('button', { name: /open modal/i });

    fireEvent.click(openModalButton);
    expect(
      screen.getAllByRole('heading', { name: mockPacientCards[0].title })
    ).toHaveLength(2);

    expect(screen.getByText(mockPacientCards[0].modal));
  });

  it('does not should render modal if no modalText is present', () => {
    const { container } = renderWithTheme(
      <Card img="teste img" title="title test" />
    );

    const openModalButton = screen.getByRole('button', { name: /open modal/i });

    fireEvent.click(openModalButton);

    const modal = screen.getByLabelText('modal');
    expect(modal).toHaveAttribute('aria-hidden', 'true');
    expect(container).toMatchSnapshot();
  });
});
