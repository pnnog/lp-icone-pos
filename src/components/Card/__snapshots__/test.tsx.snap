// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> does not should render modal if no modalText is present 1`] = `
.c0 {
  width: 100%;
  height: 100%;
}

.c1 {
  border-radius: 0.4rem;
  box-shadow: 1.8px 3.6px 1.8px rgba(0,0,0,0.25);
  position: relative;
  border: none;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.c2 {
  width: 100%;
  height: 35rem;
  border-radius: inherit;
  background: #f6f7f8;
  background-image: linear-gradient( to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100% );
  background-size: 80rem 14rem;
  -webkit-animation: placeholderShimmer 1s linear infinite forwards;
  animation: placeholderShimmer 1s linear infinite forwards;
}

.c3 {
  width: 100%;
  height: 100%;
  border-radius: inherit;
}

.c4 {
  position: absolute;
  bottom: 2rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 85%;
  word-wrap: break-word;
}

.c6 {
  margin-bottom: 2.4rem;
  font-size: 1.6rem;
  color: #F8FAFC;
  text-shadow: 0 0.3rem 0.6rem rgba(0,0,0,0.4);
}

.c7 {
  position: relative;
  padding: 0.8rem 1.6rem;
  border-radius: 0.4rem;
  text-align: center;
  border-radius: 0.4rem;
  -webkit-backdrop-filter: blur(1.5rem);
  backdrop-filter: blur(1.5rem);
  background: rgba(2,132,199,0.7);
  box-shadow: 0 0 1rem 01rem rgba(0,0,0,0.35);
  -webkit-backdrop-filter: blur(1.5rem);
}

.c7 span {
  color: #F8FAFC;
  font-weight: 600;
  font-size: 1.4rem;
}

.c8 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 40;
  overflow: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 1.6rem;
  padding-right: 2.4rem;
  -webkit-backdrop-filter: blur(1.5rem);
  backdrop-filter: blur(1.5rem);
  -webkit-backdrop-filter: blur(1.5rem);
  background: rgba(0,0,0,0.7);
  -webkit-transition: opacity 0.3s ease-out;
  transition: opacity 0.3s ease-out;
  opacity: 0;
  pointer-events: none;
}

.c9 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
  color: #F8FAFC;
  text-align: right;
}

.c9 svg {
  margin: 0.8rem;
}

.c11 {
  width: min(100%,90rem);
  margin-top: 2.4rem;
}

.c11 .c5 {
  text-shadow: none;
  position: relative;
  margin-bottom: calc(2.4rem*2);
}

.c11 .c5::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 0.6rem;
  bottom: -2rem;
  left: 0;
  background-color: #0284C7;
}

.c12 {
  font-size: 1.6rem;
  color: #F8FAFC;
  padding-bottom: 2.4rem;
  text-align: justify;
}

.c10 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (min-width:768px) {
  .c2 {
    height: 40rem;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 2.0rem;
  }
}

@media (min-width:768px) {
  .c7 {
    padding: 1.6rem 3.2rem;
  }

  .c7 span {
    font-size: 1.8rem;
  }
}

@media (min-width:768px) {
  .c8 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    background: rgba(12,74,110,.8);
  }
}

@media (min-width:1170px) {
  .c9 svg {
    margin: 2.4rem;
  }
}

@media (min-width:768px) {
  .c11 {
    margin-top: 5.6rem;
  }
}

@media (min-width:768px) {
  .c11 .c5 {
    margin-bottom: 5.6rem;
  }

  .c11 .c5::after {
    background-color: #F8FAFC;
  }
}

@media (min-width:768px) {
  .c12 {
    font-size: 1.8rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      aria-label="open modal"
      class="c1"
      role="button"
    >
      <figure
        class="c2"
      >
        <img
          alt="title test"
          class="c3"
          src="teste img"
        />
      </figure>
      <div
        class="c4"
      >
        <h2
          class="c5 c6"
        >
          title test
        </h2>
        <div
          class="c7"
        >
          <span>
             Saiba mais
          </span>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      aria-label="modal"
      class="c8"
    >
      <div
        aria-label="close modal"
        class="c9"
        role="button"
      >
        <svg
          aria-hidden="true"
          class="c10"
          fill="currentColor"
          focusable="false"
          height="30"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
          />
        </svg>
      </div>
      <div
        class="c11"
      >
        <h2
          class="c5 c6"
        >
          title test
        </h2>
        <p
          class="c12"
        />
      </div>
    </div>
  </div>
</div>
`;
